src/containers/feedOfClaim/index.tsx

```diff
@@ -164,21 +165,22 @@ const FeedClaim: React.FC<IHomeProps> = () => {
   }
 
   async function loadNextPage() {
-    if (isLastPage) return
-
-    const currentPage = Math.ceil(claimsRef.current.length / PAGE_LIMIT)
-    const nextPage = currentPage + 1
-
-    if (fetchingPage.current >= nextPage) return
-    fetchingPage.current = nextPage
+    if (isLastPage || loadingNextPage) return
 
     try {
       setLoadingNextPage(true)
-      const { data } = await fetchClaims(nextPage, searchTerm)
-      if (data.length < PAGE_LIMIT) {
+
+      // To give room for the spinner to render
+      await sleep()
```

Comment:
NO NO we do NOT Want to sleep in a responsive app please find a better way to do this.  we want to be as quick as possible.  i don't think this is a best practice.